### DISCUSSION TITLE
bug: Forward next-auth cookie session from request headers to trpc procedure

### DISCUSSION THREAD

**Discussion by jpainam** [t3-oss/create-t3-turbo#1260] [General]
### Provide environment information

N/A

### Describe the bug

A common use case, is to call an `api route`, after a `trpc` complete, for example to send an email. `API route` is usually protected with 

```tsx
// api/emails/notify.ts

const session = await auth()
if(!session){
    return new Response("Unauthorized", {status:401})
}
```

Because `request` headers are not available in the `trpc` procedure, i'm unable to forward them to the `fetch` call. Is there a better way around it.

USE case

```tsx

create: protectedProcedure
    .input(CreatePostSchema)
    .mutation(({ ctx, input }) => {
      const v = ctx.db.insert(Post).values(input);
      // WANT TO DO THIS.
      fetch(`/api/emails/notify`, {
           // i would like to forward the user session
     });
     return v;
    }),
````
Another solution will be to send this into a `redis` queue, and process these ops later. But that's not the point. I would like to make `api` calls from `trpc` procedure. This is just a simplified use case.

### Link to reproduction

https://github.com/t3-oss/create-t3-turbo

### To reproduce

N/A

### Additional information

N/A

**Comment by juliusmarminge** [comment]
just call `headers()` to get the request headers, if im not mistaken we have that on the context already though?

**Comment by jpainam** [comment]
yep, calling 
```tsx
import { headers } from "next/headers";
...
fetch(url, {
 headers: await headers()
}
```
is working.

I just thought `packages/api` needed to be free of `next.js`.
Thanks.
